src/main/kotlin/no/nav/familie/ef/iverksett/vedtakstatistikk/BehandlingDVHMapper.kt

```diff
@@ -2,62 +2,117 @@ package no.nav.familie.ef.iverksett.vedtakstatistikk
-                             behandlingId = iverksett.behandling.eksternId,
-                             relatertBehandlingId = forrigeIverksett?.behandling?.eksternId,
-                             adressebeskyttelse = iverksett.søker.adressebeskyttelse?.let { Adressebeskyttelse.valueOf(it.name) },
-                             tidspunktVedtak = iverksett.vedtak.vedtakstidspunkt.atZone(ZoneId.of("Europe/Oslo")),
-                             vilkårsvurderinger = iverksett.behandling.vilkårsvurderinger.map { mapTilVilkårsvurderinger(it) },
-                             person = mapTilPerson(personIdent = iverksett.søker.personIdent),
-                             barn = iverksett.søker.barn.map { mapTilBarn(it) },
-                             behandlingType = BehandlingType.valueOf(iverksett.behandling.behandlingType.name),
-                             behandlingÅrsak = BehandlingÅrsak.valueOf(iverksett.behandling.behandlingÅrsak.name),
-                             vedtak = Vedtak.valueOf(iverksett.vedtak.vedtaksresultat.name),
-                             vedtaksperioder = mapToVedtaksperioder(iverksett.vedtak),
-                             utbetalinger = iverksett.vedtak.tilkjentYtelse?.let {
-                                 mapTilUtbetaling(it,
-                                                  iverksett.fagsak.stønadstype,
-                                                  iverksett.fagsak.eksternId,
-                                                  iverksett.søker)
-                             } ?: emptyList(),
-                             aktivitetskrav = Aktivitetskrav(
-                                     aktivitetspliktInntrefferDato = iverksett.behandling.aktivitetspliktInntrefferDato,
-                                     harSagtOppArbeidsforhold = VilkårsvurderingUtil
-                                             .hentHarSagtOppEllerRedusertFraVurderinger(iverksett.behandling.vilkårsvurderinger)
-                             ),
-                             funksjonellId = iverksett.behandling.eksternId,
-                             stønadstype = StønadTypeEkstern.valueOf(iverksett.fagsak.stønadstype.name))
+    fun mapIverksettTilVedtakstatistikkJson(iverksett: Iverksett, forrigeIverksett: Iverksett?): String {
+        //Kunne unngått if-else på denne logikken her, men måtte unngå sealed class i kontrakt mot datavarehus
+        if (iverksett is IverksettOvergangsstønad) {
+            return mapTilVedtakOvergangsstønadDVHJson(iverksett, forrigeIverksett?.behandling?.eksternId)
+        } else if (iverksett is IverksettBarnetilsyn) {
+            return mapTilVedtakBarnetilsynDVHJson(iverksett, forrigeIverksett?.behandling?.eksternId)
+        } else {
+            error("Feil ved mapping til vedtakstatistikk: Støtter ikke skolepenger")
+        }
+    }
+
+    private fun mapTilVedtakBarnetilsynDVHJson(iverksett: IverksettBarnetilsyn, forrigeIverksettBehandlingEksternId: Long?): String {
```

Comment:
For å enklere kunne teste og tydligere API, hadde man kunnet la denne returnere`BehandlingBarnetilsynDVH` og at `VedtakstatistikkKafkaProducer->sendVedtak` har 2 metoder, en for hver objekt som skal sendes til DVH, og at disse sen kaller en private metode i `VedtakstatistikkKafkaProducer` 

```kotlin
fun sendVedtak(vedtakStatistikk: BehandlingDVH) {
  sendVedtak(vedtakStatistikk.behandlingId.toString(), vedtakStatistikk.toJson())
}

private fun sendVedtak(behandlingId: String, vedtakStatistikk: String) {
```

Reply:
Ikke uenig den! Fixed.